plugins/zenoh-plugin-trait/src/manager.rs

```diff
@@ -155,38 +170,46 @@ impl<StartArgs: PluginStartArgs + 'static, Instance: PluginInstance + 'static>
         tracing::debug!("Declared dynamic plugin {} by name {}", &name, &plugin_name);
         let loader = DynamicPlugin::new(
             name,
+            id.clone(),
             DynamicPluginSource::ByName((libloader, plugin_name)),
             required,
         );
-        self.plugins.push(PluginRecord::new(loader));
-        Ok(self.plugins.last_mut().unwrap())
+        self.plugins.insert(id.clone(), PluginRecord::new(loader));
```

Comment:
No protection against adding plugin with the same name. Before it was wrong also: same-named plugin was added, but it wasn't possible to access it by name later. Now the old one is just silently removed